Description: only emit warnings about overriding LSB default with symlink
 state for the scripts in insserv arg list. Use "overrides" instead of
 "overwrites" and define defaults as "LSB defaults".
From: Kel Modderman <kel@otaku42.de>
Bug-Debian: bugs.debian.org/668559
---
--- a/insserv.c
+++ b/insserv.c
@@ -3081,7 +3081,7 @@ int main (int argc, char *argv[])
 			     */
 			    if (!defaults && (deflvls != service->start->lvl)) {
 				if (!del && isarg && !(argr[curr_argc]))
-				    warn("warning: current start runlevel(s) (%s) of script `%s' overwrites defaults (%s).\n",
+				    warn("warning: current start runlevel(s) (%s) of script `%s' overrides LSB defaults (%s).\n",
 					 service->start->lvl ? lvl2str(service->start->lvl) : "empty", d->d_name, lvl2str(deflvls));
 			    }
 			} else
@@ -3100,8 +3100,9 @@ int main (int argc, char *argv[])
 			     * of the current script.
 			     */
 			    if (!defaults && service->start->lvl != 0) {
-				warn("warning: current start runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
-				     lvl2str(service->start->lvl), d->d_name);
+				if (!del && isarg && !(argr[curr_argc]))
+				    warn("warning: current start runlevel(s) (%s) of script `%s' overrides LSB defaults (empty).\n",
+					 lvl2str(service->start->lvl), d->d_name);
 				script_inf.default_start = lvl2str(service->start->lvl);
 			    }
 			}
@@ -3143,7 +3144,7 @@ int main (int argc, char *argv[])
 			     */
 			    if (!defaults && (deflvlk != service->stopp->lvl)) {
 				if (!del && isarg && !(argr[curr_argc]))
-				    warn("warning: current stop runlevel(s) (%s) of script `%s' overwrites defaults (%s).\n",
+				    warn("warning: current stop runlevel(s) (%s) of script `%s' overrides LSB defaults (%s).\n",
 					 service->stopp->lvl ? lvl2str(service->stopp->lvl) : "empty", d->d_name, lvl2str(deflvlk));
 			    }
 			} else
@@ -3162,8 +3163,9 @@ int main (int argc, char *argv[])
 			     * of the current script.
 			     */
 			    if (!defaults && service->stopp->lvl != 0) {
-				warn("warning: current stop runlevel(s) (%s) of script `%s' overwrites defaults (empty).\n",
-				     lvl2str(service->stopp->lvl), d->d_name);
+				if (!del && isarg && !(argr[curr_argc]))
+				    warn("warning: current stop runlevel(s) (%s) of script `%s' overrides LSB defaults (empty).\n",
+					 lvl2str(service->stopp->lvl), d->d_name);
 				script_inf.default_stop = lvl2str(service->stopp->lvl);
 			    }
 			}
